tests/unit/listObjectsByReplicationStatus.js

```diff
@@ -0,0 +1,85 @@
+const { listObjectsByReplicationStatus } = require('../../listObjectsByReplicationStatus');
+
+describe('listObjectsByReplicationStatus - Input Validation', () => {
+    const validOptions = {
+        buckets: 'test-bucket',
+        accessKey: 'test-access-key',
+        secretKey: 'test-secret-key',
+        endpoint: 'http://localhost:8000',
+        replicationStatus: 'PENDING',
+    };
+
+    it('should reject when buckets is missing', async () => {
+        const options = { ...validOptions, buckets: null };
+        await expect(listObjectsByReplicationStatus(options))
+            .rejects
+            .toThrow('No buckets given as input! Please provide a comma-separated list of buckets');
+    });
+
+    it('should reject when buckets is empty string', async () => {
+        const options = { ...validOptions, buckets: '' };
+        await expect(listObjectsByReplicationStatus(options))
+            .rejects
+            .toThrow('No buckets given as input! Please provide a comma-separated list of buckets');
+    });
+
+    it('should reject when buckets is only whitespace', async () => {
+        const options = { ...validOptions, buckets: '   ' };
+        await expect(listObjectsByReplicationStatus(options))
+            .rejects
+            .toThrow('No buckets given as input! Please provide a comma-separated list of buckets');
+    });
+
+    it('should reject when endpoint is missing', async () => {
+        const options = { ...validOptions, endpoint: null };
```

Comment:
nitpicking: I'd rather test when the `endpoint` attribute is completely missing (probably a more common error). Same for the rest of the attributes below.